src/base/bittorrent/sessionstatus.h

```diff
@@ -29,50 +29,43 @@
 #ifndef BITTORRENT_SESSIONSTATUS_H
 #define BITTORRENT_SESSIONSTATUS_H
 
-#include <libtorrent/session_status.hpp>
 #include <QtGlobal>
 
 namespace BitTorrent
 {
-    class SessionStatus
+    struct SessionStatus
     {
-    public:
-        SessionStatus(const libtorrent::session_status &nativeStatus);
```

Comment:
>while in the second case you see that the whole function is valid only under certain condition.

This is one such case! `libtorrent::session_stats_alert` does not exist in libtorrent-1.0 so appropriate functions are valid only if libtorrent >= 1.1.